plotly/tests/test_optional/test_opt_tracefactory.py

```diff
@@ -105,3 +105,54 @@ def test_simple_streamline(self):
         self.assertListEqual(strln['data'][0]['x'][0:100],
                              expected_strln_0_100['x'])
 
+
+class TestDendrogram(TestCase):
+
+    def test_default_dendrogram(self):
+        dendro = tls.FigureFactory.create_dendrogram(X=np.array([[1, 2, 3, 4],
+                                                                 [1, 1, 3, 4],
+                                                                 [1, 2, 1, 4],
+                                                                 [1, 2, 3, 1]]))
+        expected_dendro_data = [{'marker': {'color': 'rgb(255,133,27)'},
+                                 'mode': 'lines', 'xaxis': 'xs',
+                                 'yaxis': 'ys',
```

Comment:
These 'yaxis' and 'xaxis' strings are off. I think it's because your default string is `'xaxis'` and you use `xaxis[-1]` which ends up being the string `'s'`. These should probably default to `'x'` and `'y'`, right @chriddyp?